parser/src/lexer/tests/mod.rs

```diff
@@ -17,21 +15,65 @@ mod variables;
 // TEST HELPERS
 // ================================================================================================
 
+use std::sync::Arc;
+
+use miden_diagnostics::CodeMap;
+use miden_parsing::{FileMapSource, Scanner, Source};
+
 fn expect_valid_tokenization(source: &str, expected_tokens: Vec<Token>) {
-    let tokens: Vec<Token> = Lexer::new(source).collect();
+    let codemap = Arc::new(CodeMap::new());
+    let id = codemap.add("nofile", source.to_string());
+    let file = codemap.get(id).unwrap();
+    let scanner = Scanner::new(FileMapSource::new(file));
+    let lexer = Lexer::new(scanner);
+
+    let tokens: Vec<Token> = lexer.map(|res| res.unwrap().1).collect();
     assert_eq!(tokens, expected_tokens);
 }
 
-/// Checks that source is valid and asserts that appropriate error is returned if there
-/// is a problem while scanning the source.
-fn expect_error(source: &str, error: Error) {
-    let mut tokens = Lexer::new(source)
-        .spanned()
-        .map(Token::to_spanned)
-        .peekable();
+/// Asserts that lexing fails with a specific error, that occurs at the given line and column.
+///
+/// NOTE: Provided line/column numbers should be zero-indexed
+fn expect_error_at_location(source: &str, expected: LexicalError, line: u32, col: u32) {
+    use miden_diagnostics::{ColumnIndex, LineIndex};
+
+    let codemap = Arc::new(CodeMap::new());
+    let mut tokens = lex(codemap.clone(), source);
+    let err = tokens
+        .find_map(|res| match res {
+            Ok(_) => None,
+            Err(ParseError::Lexer(err)) => Some(err),
+            Err(_) => unreachable!(),
+        })
+        .expect("expected lexical error, but lexing completed successfully");
 
-    while tokens.next_if(|token| token.is_ok()).is_some() {}
-    let err = tokens.next();
+    let loc = match &err {
+        LexicalError::InvalidInt { span, .. } => codemap.location_for_span(*span).unwrap(),
```

Comment:
it seems this broke `next`:

```
  --> parser/src/lexer/tests/mod.rs:51:58
   |
51 |         LexicalError::InvalidInt { span, .. } => codemap.location_for_span(*span).unwrap(),
   |                                                          ^^^^^^^^^^^^^^^^^ method not found in `Arc<CodeMap>`
```

Reply:
It should be `location` rather than 'location_from_span` now, that should fix everything right up

Reply:
That one should be changed to `location_at_index`, same arguments

Reply:
`location_at_index` needs a `SourceId`, the error `UnexpectedCharacter` doesn't have one, it is defined as:

```
UnexpectedCharacter { start: SourceIndex, found: char },
```

From what I could tell the `SourceIndex` is a byte-offset inside a file, but the code expects a File identifier of sorts.

Reply:
It has a `source_id` method for the SourceId

Reply:
please review: https://github.com/0xPolygonMiden/air-script/pull/298